### PR TITLE
feat: Added scale function in paddle frontend

### PR DESCRIPTION
**Pull request by Aaryan562**
Resolves  https://github.com/unifyai/ivy/issues/23297

### CODE REVIEW

**Review by HaiderSultanArc — COMMENTED** [review]
LGTM but CI is not working on your tests. Let's wait till it's running again!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py [hunk 1]
```diff
@@ -2061,6 +2061,43 @@ def test_paddle_rsqrt(
...
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        backend_to_test=backend_fw,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        x=x[0],
+        scale=scale,
+        bias=bias,
+        bias_after_scale=bias_after_scale,
```
Comment (HaiderSultanArc) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py:2097]: add `rtol=0.000001` after this argument.

Comment (Aaryan562) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py:2098]: I have added rtol, but it failed at torch framework for the datatype int16 when it took values like  (['int16'], [array(0, dtype=int16)]),
scale=1.9999999701976776,
bias=1.0,
which returned value 2 but the groundtruth is 1 but if we lower the decimal values then we get the same value as that of the groudtruth

Comment (Aaryan562) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py:2098]: Basically I have tested for each backend multiple datatypes which the scale supports and here are the results for it with values x=[[0]],
scale=1.9999999701976776
bias=1.0
-----------------numpy---------------------------
value of gd for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of backend for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of gd for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of backend for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of gd for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
[2023-09-29 15:00:55,613] [ WARNING] xla_bridge.py:636 - No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
/workspaces/ivy/ivy/utils/exceptions.py:390: UserWarning: The current backend: 'jax' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
  warnings.warn(
-----------------jax---------------------------
value of gd for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of backend for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of gd for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of backend for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of gd for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
/workspaces/ivy/ivy/utils/exceptions.py:390: UserWarning: The current backend: 'tensorflow' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
  warnings.warn(
-----------------tensorflow---------------------------
value of gd for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of backend for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of gd for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of backend for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of gd for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
/workspaces/ivy/ivy/utils/exceptions.py:390: UserWarning: The current backend: 'paddle' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
  warnings.warn(
-----------------paddle---------------------------
value of gd for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of backend for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of gd for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of backend for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of gd for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
-----------------torch---------------------------
value of gd for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for uint8 is Tensor(shape=[1, 1], dtype=uint8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int16 is Tensor(shape=[1, 1], dtype=int16, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int8 is Tensor(shape=[1, 1], dtype=int8, place=Place(cpu), stop_gradient=True,
       [[1]])
value of gd for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of backend for float32 is Tensor(shape=[1, 1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [[2.]])
value of gd for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of backend for float64 is Tensor(shape=[1, 1], dtype=float64, place=Place(cpu), stop_gradient=True,
       [[1.99999997]])
value of gd for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])
value of backend for int32 is Tensor(shape=[1, 1], dtype=int32, place=Place(cpu), stop_gradient=True,
       [[1]])

Comment (Aaryan562) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py:2098]: import paddle
import ivy.functional.frontends.paddle as paddle_bc
import ivy
import jax
jax.config.update('jax_enable_x64', True)
for backend in ['numpy','jax','tensorflow','paddle','torch']:
    ivy.set_backend(backend)
    print(f"-----------------{backend}---------------------------")
    for dtype in ['uint8','int16','int8','float32','float64','int32']:
        data=paddle.to_tensor([[0]], dtype=dtype)
        res = paddle.scale(data, scale=1.9999999701976776, bias=1.0, bias_after_scale=False)
        print(f"value of gd for {dtype} is {res}")
        data_=paddle_bc.to_tensor([[0]], dtype=dtype)
        res_bc=paddle_bc.scale(data_, scale=1.9999999701976776, bias=1.0, bias_after_scale=False)
        print(f"value of backend for {dtype} is {res}")
    ivy.unset_backend()
   The above is the code to test manually.

Comment (Aaryan562) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py:2098]: image

**Review by HaiderSultanArc — CHANGES_REQUESTED** [review]
Merge upstream/main into your branch

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py (hunk 1 above)
Comment (HaiderSultanArc) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py:2098]: Show me the example that is failing 🤔 which decimal value needs to be lowered?

Comment (Aaryan562) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py:2098]: > Show me the example that is failing 🤔 which decimal value needs to be lowered

When I tested manually the ground truth and backend values match for the example I shown above with the output as well. When I used the hypothesis search then the values which are similar to thr example tested manually it fails. I honestly dont know how to proceed forward. I tried everything and again I come to the conclusion if the precision of the arguments are more than 6-7 decimal points then it fails for certain datatypes and for certain backends. I rest my case upto you now😞

**Comment by Aaryan562** [comment]
> Merge upstream/main into your branch

done

**Comment by HaiderSultanArc** [comment]
@Aaryan562 your tests are failing for multiple backends. You have to fix your implementation.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py (hunk 1 above)
Comment (HaiderSultanArc) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_math.py:2098]: CI tests are failing. Could you make sure your implementation is correct? It's a value different not just the 6th or 7th decimal place it's an entirely different number you are getting.

**Comment by vedpatwardhan** [comment]
actually the problem is with the `scale` and `bias` arguments as there's no bound to the precision (64 bits by default) which leads to a large-enough error to fail the tests when the dtype of `x` is an integer. There's a `floats` helper in the tests which enables adding such bounds as a result of which simply replacing `st.floats` by `helpers.floats` should get the tests passing (for backends other than `tensorflow` which is failing due to the `KeyError` which is being fixed on `main` currently) 😄

**Comment by HaiderSultanArc** [comment]
Thanks @vedpatwardhan 😄 . I didn't know about this. I will look at the helper functions in more depth!

**Review by HaiderSultanArc — COMMENTED** [review]
CI isn't picking up your tests here either 🤔

**Comment by vedpatwardhan** [comment]
> CI isn't picking up your tests here either 🤔

@RashulChutani could you please take a look at this PR as well? Thanks 😄

**Review by HaiderSultanArc — CHANGES_REQUESTED** [review]
@Aaryan562 please merge upstream/main into your branch

**Pull request closed without merging**